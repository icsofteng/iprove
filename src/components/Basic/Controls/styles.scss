.controlList {
  width: 10%;
  position: absolute;
  top: 0;
  right: 0;
  background: #e0e0e0;
  display: flex;
  flex-direction: column;
  height: 100%;
}
.controlBlock {
  cursor: pointer;
  display: flex;
  align-items: center;
  background: #0e80c5;
  padding: 15px;
  margin-bottom: 25px;
}
.controlBlock:hover {
  opacity: 0.6;
}
.label {
  font-size: 15px;
  color: #fff;
  font-weight: bold;
  margin-right: 10px;
}
.template {
  display: flex;
}
.exprPlaceholder {
  display: block;
  width: 20px;
  height: 20px;
  background: #02466f;
  margin-right: 4px;
}
.symbolPlaceholder {
  display: flex;
  height: 20px;
  background: #ffffff;
  margin-right: 4px;
  text-align: center;
  font-size: 12px;
  font-weight: bold;
  align-items: center;
  padding-left: 7px;
  padding-right: 7px;
}
