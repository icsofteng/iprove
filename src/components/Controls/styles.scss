:global body {
  margin:0;
  font-family: 'Helvetica Neue', 'Segoe UI', Arial, Helvetica, sans-serif
}
.title {
  display: inline-block;
  margin: 0;
  font-size: 27px;
}
.controls {
  display: block;
  background: #ebebeb;
  padding: 20px;
}
.controlList {
  margin: 30px 0 10px;
  display: flex;
}
.controlBlock {
  border: 1px solid #d1d1d1;
  padding: 10px;
  width: 90px;
  margin-right: 15px;
  cursor: pointer;
}
.controlBlock:hover {
  opacity: 0.6;
}
.label {
  font-size: 12px;
  color: #333;
  font-weight: bold;
  position: absolute;
  background: #ebebeb;
  margin: -17px 0;
}
.template {
  display: flex;
  margin: 8px 1px 0;
}
.exprPlaceholder {
  display: block;
  width: 15px;
  height: 15px;
  background: #8a8a8a;
  margin-right: 4px;
}
.symbolPlaceholder {
  display: block;
  width: 15px;
  height: 15px;
  background: #ffffff;
  margin-right: 4px;
  text-align: center;
  font-size: 12px;
  font-weight: bold;
}
