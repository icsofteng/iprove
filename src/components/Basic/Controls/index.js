import React from 'react'
import cx from 'classnames'
import styles from './styles.scss'

const ControlBlock = ({ type, label, symbol, exprLeft=false, exprRight }) =>
  <div data-type={type} className={cx('drag-drop', styles.controlBlock)}>
    <div className={styles.label}>{label}</div>
    <div className={styles.template}>
      { exprLeft && <div className={styles.exprPlaceholder}></div> }
      <div className={styles.symbolPlaceholder}>{symbol}</div>
      { exprRight && <div className={styles.exprPlaceholder}></div> }
    </div>
  </div>

const Controls = (props) =>
  <div className={styles.controlList}>
    <ControlBlock type="binary" label="Binary Rule" symbol="?" exprLeft exprRight {...props} />
    <ControlBlock type="unary" label="Unary Rule" symbol="?" exprRight {...props}/>
    <ControlBlock type="identifier" label="Literal" symbol="P" {...props}/>
    <ControlBlock type="true" label="True" symbol="⊤" {...props}/>
    <ControlBlock type="false" label="False" symbol="⊥" {...props}/>
  </div>

export default Controls
