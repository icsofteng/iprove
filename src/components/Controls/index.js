import React from 'react'
import { connect } from 'react-redux'
import styles from './styles.scss'
import { CLICK_SYMBOL } from '../../constants'

// const controls = ['∧', '∨', '¬', '⇒', '⇐', '⇔', '⊤', '⊥']
const Controls = (props) =>
  <div className={styles.controls}>
    <h1 className={styles.title}>iProve</h1>
    <div className={styles.controlList}>
      
      <div className={styles.controlBlock}>
        <div className={styles.label}>Binary Rule</div>
        <div className={styles.template}>
          <div className={styles.exprPlaceholder}></div>
          <div className={styles.symbolPlaceholder}>?</div>
          <div className={styles.exprPlaceholder}></div>
        </div>
      </div>

      <div className={styles.controlBlock}>
        <div className={styles.label}>Unary Rule</div>
        <div className={styles.template}>
          <div className={styles.symbolPlaceholder}>?</div>
          <div className={styles.exprPlaceholder}></div>
        </div>
      </div>

      <div className={styles.controlBlock}>
        <div className={styles.label}>Literal</div>
        <div className={styles.template}>
          <div className={styles.symbolPlaceholder}>x</div>
        </div>
      </div>

      <div className={styles.controlBlock}>
        <div className={styles.label}>True</div>
        <div className={styles.template}>
          <div className={styles.symbolPlaceholder}>⊤</div>
        </div>
      </div>

      <div className={styles.controlBlock}>
        <div className={styles.label}>False</div>
        <div className={styles.template}>
          <div className={styles.symbolPlaceholder}>⊥</div>
        </div>
      </div>

    </div>
  </div>

const mapDispatchToProps = dispatch => {
  return {
    clickSymbol: (symbol) => dispatch({ type: CLICK_SYMBOL, payload: symbol })
  }
}

export default connect(null, mapDispatchToProps)(Controls)