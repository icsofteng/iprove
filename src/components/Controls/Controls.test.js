import React from 'react'

import Controls from '.'



