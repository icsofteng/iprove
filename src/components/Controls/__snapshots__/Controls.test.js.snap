// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div
  className="controls"
>
  <h1
    className="title"
  >
    iProve
  </h1>
  <div
    className="controlList"
  >
    <div
      className="drag-drop controlBlock"
      data-type="binary"
    >
      <div
        className="label"
      >
        Binary Rule
      </div>
      <div
        className="template"
      >
        <div
          className="exprPlaceholder"
        />
        <div
          className="symbolPlaceholder"
        >
          ?
        </div>
        <div
          className="exprPlaceholder"
        />
      </div>
    </div>
    <div
      className="drag-drop controlBlock"
      data-type="unary"
    >
      <div
        className="label"
      >
        Unary Rule
      </div>
      <div
        className="template"
      >
        <div
          className="symbolPlaceholder"
        >
          ?
        </div>
        <div
          className="exprPlaceholder"
        />
      </div>
    </div>
    <div
      className="drag-drop controlBlock"
      data-type="literal"
    >
      <div
        className="label"
      >
        Literal
      </div>
      <div
        className="template"
      >
        <div
          className="symbolPlaceholder"
        >
          x
        </div>
      </div>
    </div>
    <div
      className="drag-drop controlBlock"
      data-type="true"
    >
      <div
        className="label"
      >
        True
      </div>
      <div
        className="template"
      >
        <div
          className="symbolPlaceholder"
        >
          ⊤
        </div>
      </div>
    </div>
    <div
      className="drag-drop controlBlock"
      data-type="false"
    >
      <div
        className="label"
      >
        False
      </div>
      <div
        className="template"
      >
        <div
          className="symbolPlaceholder"
        >
          ⊥
        </div>
      </div>
    </div>
  </div>
</div>
`;
